apiserver/facades/client/application/application.go

```diff
@@ -518,7 +518,7 @@ func deployApplication(
 
 	// This check is done early so that errors deeper in the call-stack do not
 	// leave an application deployment in an unrecoverable error state.
-	if err := checkMachinePlacement(backend, args); err != nil {
+	if err := checkMachinePlacement(backend, args.ApplicationName, args.Placement); err != nil {
```

Comment:
I'd tend not to expand this, fewer argument being better; also less likely to require change in future.